R/surv-brier.R

```diff
@@ -0,0 +1,164 @@
+#' Brier score for right censored data
+#'
+#' Compute the time-dependent Brier score for right censored data. Which is the
+#' mean squared error at time point `.time`.
+#'
+#' @family dynamic survival metrics
+#' @templateVar fn brier_survival
+#' @template return-dynamic-survival
+#' @details
+#'
+#' Smaller values of the score are associated with better model performance.
+#'
+#' @inheritParams pr_auc
+#'
+#' @param data A `data.frame` containing the columns specified by `truth` and
+#' `estimate`.
+#'
+#' @param truth The column identifier for the true class survival result (that
+#' is created using [survival::Surv()].). This should be an unquoted column name
+#' although this argument is passed by expression and supports
+#' [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#' `_vec()` functions, an [survival::Surv()] object.
+#'
+#' @param estimate The column identifier for the survival probabilities. This
+#' should be a numeric vector. This should be an unquoted column name although
+#' this argument is passed by expression and supports
+#'  [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#' `_vec()` functions, a numeric vector.
+#'
+#' @param censoring_weights The column identifier for censoring weights. This is
+#' expected to a numeric vector. This should be an unquoted column name although
+#' this argument is passed by expression and supports
+#' [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#' `_vec()` functions, a numeric vector.
+#'
+#' @param .time The column identifier for the time point. This
+#' should be a numeric vector, with 1 unique value for each group. This should
+#' be an unquoted column name although this argument is passed by expression and
+#' supports [quasiquotation][rlang::quasiquotation] (you can unquote column
+#' names). For `_vec()` functions, a numeric vector.
+#'
+#' @param ... Not currently used.
+#'
+#' @author Emil Hvitfeldt
+#'
+#' @references
+#'   E. Graf, C. Schmoor, W. Sauerbrei, and M. Schumacher, “Assessment and
+#'   comparison of prognostic classification schemes for survival data,”
+#'   Statistics in Medicine, vol. 18, no. 17-18, pp. 2529–2545, 1999.
+#'
+#' @examples
+#' library(dplyr)
+#'
+#' lung_surv %>%
+#'   group_by(.time) %>%
+#'   brier_survival(
+#'     truth = surv_obj,
+#'     estimate = .pred_survival,
+#'     censoring_weights = prob_censored,
+#'     .time = .time
+#'   )
+#' @export
+brier_survival <- function(data, ...) {
+  UseMethod("brier_survival")
+}
+
+brier_survival <- new_dynamic_survival_metric(
+  brier_survival,
+  direction = "minimize"
+)
+
+#' @rdname brier_survival
+#' @export
+brier_survival.data.frame <- function(data,
+                                      truth,
+                                      estimate,
+                                      censoring_weights,
+                                      .time,
```

Comment:
i suggest just calling this `time`. only use `.` as a prefix if you absolutely have to because it would conflict with inputs the users can type in `...`

Reply:
we are going to change it everywhere anyway once we find a name. so i'm just keeping it consistent right now to make find-replace easy